src/events.rs

```diff
@@ -145,6 +147,22 @@ macro_rules! make_pointer_event {
     };
 }
 
+macro_rules! make_touch_event {
+    ($name:ident, $type:literal) => {
+        make_event!($name, $type => web_sys::TouchEvent);
+
+        impl $name {
+            #[inline] pub fn ctrl_key(&self) -> bool { self.event.ctrl_key() || self.event.meta_key() }
+            #[inline] pub fn shift_key(&self) -> bool { self.event.shift_key() }
+            #[inline] pub fn alt_key(&self) -> bool { self.event.alt_key() }
+    
+            #[inline] pub fn changed_touches(&self) -> TouchList { self.event.changed_touches() }
```

Comment:
Maybe we can do better here and return a `Vec<Touch>`? `TouchList` is really annoying to work with, it's a bad user experience.

Reply:
Actually, I think we can do even better and return an `impl Iterator<Item = Touch>`, that should be more efficient.

Reply:
Something like this should work:

```rust
#[derive(Debug)]
struct TouchListIter {
    list: TouchList,
    index: u32,
    length: u32,
}

impl TouchListIter {
    fn new(list: TouchList) -> Self {
        Self {
            index: 0,
            length: list.length(),
            list,
        }
    }
}

impl Iterator for TouchListIter {
    type Item = Touch;

    fn next(&mut self) -> Option<Self::Item> {
        let index = self.index;

        if index < self.length {
            self.index += 1;
            self.list.get(index)

        } else {
            None
        }
    }
}
```

Reply:
Nice idea! I added the code as suggested.